Automattic/woocommerce-payments
# Send complete store_api_url to platform site

This ensures the URL obeys the merchant's permalink preference.

Fixes platform checkout issue: 750
Related platform checkout PR: 908

#### Changes proposed in this Pull Request
Send the complete merchant's store API URL taking into account the permalink preferences.
 








#### Testing instructions





* Add a breakpoint to this line https://github.com/Automattic/woocommerce-payments/blob/95145fdda3ba121b790b363ed41d72864c2ac4a3/includes/class-wc-payments.php#L1011
* As a merchant, select WP permalink structure as "Post Name" from Settings -> Permalinks.
* As a customer, go through the checkout process with a registered platform checkout email address.
* After completing the OTP verification, notice that the `store_api_url` sent to platform site is in the `https://MERCHANT.SITE/wp-json/wc/store` format
* As a merchant, change WP permalink structure to "Plain" from Settings -> Permalinks.
* Go through platform checkout again, notice that the `store_api_url` sent to platform site is in the `https://MERCHANT.SITE/index.php?rest_route=/wc/store` format

-------------------


**Post merge**
